subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/DefaultToolchainSpec.java

```diff
@@ -107,6 +108,14 @@ public boolean isConfigured() {
 
     @Override
     public boolean isValid() {
+        if (vendor.getOrNull() == JvmVendorSpec.IBM_SEMERU) {
+            // https://github.com/gradle/gradle/issues/23155
+            // This should make the spec invalid when the enum gets removed
+            DeprecationLogger.deprecateBehaviour("Using JVM vendor IBM_SEMERU.")
```

Comment:
Maybe "requesting" instead of "using" since you're still able to use a JDK from IBM?
```suggestion
            DeprecationLogger.deprecateBehaviour("Requesting JVM vendor IBM_SEMERU.")
```